app/bundles/LeadBundle/Entity/TagRepository.php

```diff
@@ -74,4 +74,34 @@ public function getTagsByName($tags)
 
         return $results;
     }
+
+    /**
+     * Check Lead tags by Ids.
+     *
+     * @param Lead $lead
+     * @param $tags
+     *
+     * @return bool
+     */
+    public function checkLeadByTags(Lead $lead, $tags)
+    {
+        if (empty($tags)) {
+            return false;
+        }
+
+        $q = $this->_em->getConnection()->createQueryBuilder();
+        $q->select('l.id')
+            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
+        $q->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
+                ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
+                ->where(
+                    $q->expr()->andX(
+                        $q->expr()->in('t.tag', "'".implode("','", $tags)."'"),
```

Comment:
How about if you pass 3rd argument as `[\Doctrine\DBAL\Connection::PARAM_INT_ARRAY]`? I don't have experience with that, but it should work somehow like that according to docs: http://docs.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/data-retrieval-and-manipulation.html#list-of-parameters-conversion